File: ts/examples/simple-example-ts/src/data_types/emailDataType.ts
import {Identifier, Name, Schema} from "@code0-tech/hercules";
import {z} from "zod";

const EmailSchema = z.string().regex(/^[^@]+@[^@]+\.[^@]+$/)
export type EmailType = z.infer<typeof EmailSchema>

@Identifier("email_address")
@Name({code: "en-US", content: "Email Address"})
@Schema(EmailSchema)
export class EmailDataType {}

File: py/examples/simple-example-py/data_types/email_data_type.py
from hercules import Identifier, Name, Schema
from hercules.schema import RootModel, Annotated, StringConstraints


class EmailModel(RootModel[Annotated[str, StringConstraints(pattern=r"^[^@]+@[^@]+\.[^@]+$")]]):
    """An email address, described as a constrained string."""


@Identifier("email_address")
@Name({"code": "en-US", "content": "Email Address"})
@Schema(EmailModel)
class EmailDataType:
    pass
